architecture/consistency.md

```diff
@@ -0,0 +1,96 @@
+
+#### No distributed transactions
+
+Distributed transactions (transactions that span multiple services, e.g. a database and a message queue) are required to ensure strong consistency of all operations within the shared transaction. On Windows, the Distributed Transaction Coordinator (DTC) is able coordinate transactions across multiple compatible participants using two-phase commit protocols. Each participating resource of a distributed transaction must explicitly support the two-phase commit protocol. While traditional on-premises focused services like MSMQ or MSSQL server support the DTC, managed cloud services do not support DTC transactions, enforcing different strategies to achieve consistency across transactional resources.
+
+### Outbox Pattern
+
+The [transactional Outbox Pattern](https://microservices.io/patterns/data/transactional-outbox.html) is a pattern to achieve atomically consitent database and message queueing operations. Due to the lack of distributed transactions in many modern application environments, the outbox pattern is a commonly used pattern to solve strong consistency requirements.
+
+The outbox pattern is implemented by storing outgoing messages into the same database as the business operations. This allows the use of database native transaction capabilities to achieve atomic consistency. In a second step, the persisted messages are dispatched to tbe message queue.
+
+The nature of the outbox pattern might cause outgoing messages to be dispatched multiple times. This should be taken into consideration by leveraging [idempotency mechanisms](#idempotency) on the receiver.
+
+Note: Implementing the outbox pattern is very risky and error-prone. Small mistakes can lead to unintended behavior and message or data loss. Consider the the [NServiceBus Outbox feature](/nservicebus/outbox/) which implements the Outbox pattern, including built-in message deduplication when receiving messages. The NServiceBus Outbox is used in production by hundreds of customers, thoroughly tested and well documented.
+
+
+#### Transactional Session
+
+The TransactionalSession is an NServiceBus feature specifically designed to bring Outbox functionality to system boundaries like web servers that require outgoing message operations and database operations to be consistent.
+
+### Session Management
+
+When NServiceBus is configured with a persistent storage, NServiceBus will manage the lifecycle of (transactional) storage sessions. Application code can access the managed session to work with business data:
+
+```mermaid
+sequenceDiagram
+    Queue->>NSB: ReceiveMessage
+    NSB->>Storage: Open session
+    NSB->>Application: Invoke message handler
+    Application->>Storage: Store business data
+    Application-->>NSB: message handler finished
+    NSB->>Storage: Store workflow/outbox data
+    NSB->>Storage: Commit transaction
+    NSB-xStorage: Close session
+    NSB-->>Queue: Ack message
+```
```

Comment:
```suggestion
```

I think this might not be valuable enough from an architectural perspective and is too NSB specific for this section.

Reply:
Is this anywhere else?

Reply:
no but we could consider adding this to the saga documentation instead.

Reply:
As discussed, we'll remove this from this PR and raise an issue instead.